### PR TITLE
BigBed-parsing improvements

### PR DESCRIPTION
**Pull request by ryan-williams**
- handle non-compressed BigBed files
- allow versions < 4
- explain mysterious "+ 2 offset" comment
- a few other nits

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/pileup.js/427)

### CODE REVIEW

File: src/main/data/BigBed.js [hunk 1]
```diff
@@ -53,13 +53,21 @@ function reverseContigMap(contigMap: {[key:string]: number}): Array<string> {
...
+function extractFeaturesFromBlock(buffer, dataRange, block, isCompressed): ChrIdBedRow[] {
   var blockOffset = block.offset - dataRange.start,
       blockLimit = blockOffset + block.size,
-      // TODO: where does the +2 come from? (I copied it from dalliance)
-      blockBuffer = buffer.slice(blockOffset + 2, blockLimit);
-  // TODO: only inflate if necessary
-  var inflatedBuffer = pako.inflateRaw(new Uint8Array(blockBuffer));
+
+      blockBuffer =
+        // NOTE: "+ 2" skips over two bytes of gzip header (0x8b1f), which pako.inflateRaw will not handle.
```
Comment (armish) [inline src/main/data/BigBed.js]: clarification much appreciated ;)

File: src/test/data/BigBed-test.js [hunk 2]
```diff
@@ -43,6 +47,28 @@ describe('BigBed', function() {
...
+  it('should extract features from an uncompressed BigBed', function () {
+    var bb = getUncompressedTestBigBed();
+
+    return bb.getFeaturesInRange('chr17', 60000, 270000)
+      .then(features => {
+        // Here's what these three lines in the file look like:
+        // chr17	62296	202576
+        // chr17	62296	202576
+        // chr17	260433	264713
+        expect(features).to.have.length(3);
```
Comment (armish) [inline src/test/data/BigBed-test.js]: you can shorten this as:

```javascript

expect(features).to.deep.equal(
  [{contig: 'chr17', start: 62296, stop: 202576},
   {contig: 'chr17', start: 62296, stop: 202576},
   {contig: 'chr17', start: 260433, stop: 264713}]);
```

File: src/main/data/BigBed.js [hunk 3]
```diff
@@ -53,13 +53,21 @@ function reverseContigMap(contigMap: {[key:string]: number}): Array<string> {
 }
 
 
-function extractFeaturesFromBlock(buffer, dataRange, block): ChrIdBedRow[] {
+function extractFeaturesFromBlock(buffer, dataRange, block, isCompressed): ChrIdBedRow[] {
```
Comment (armish) [inline src/main/data/BigBed.js]: would you mind annotating these parameters with flow types while at it? If they are already obvious, feel free to ignore this comment ;)

**Comment by armish** [comment]
@ryan-williams: LGTM. Have a minor comment that might or might not apply to your case.

👍

File: src/main/data/BigBed.js (hunk 3 above)
Comment (ryan-williams) [inline src/main/data/BigBed.js]: they were not obvious; done!

File: src/test/data/BigBed-test.js (hunk 2 above)
Comment (ryan-williams) [inline src/test/data/BigBed-test.js]: thx, done

**Comment by ryan-williams** [comment]
all addressed, thx @armish

**Comment by armish** [comment]
thanks @ryan-williams! Will merge on green :+1:

**Pull request merged**